### PR TITLE
Add validation configuration

### PR DESCRIPTION
**Pull request by xichen2020**
cc @dgromov @cw9 @jeromefroe 

This PR pulls in the latest m3metrics and sets up the validation configuration to perform validation on rule changes.

### CODE REVIEW

File: services/r2ctl/config/r2ctl.go [hunk 1]
```diff
@@ -52,31 +54,41 @@ type Configuration struct {
...
 	Auth *auth.SimpleAuthConfig `yaml:"auth"`
-
-	// Clock configuration used by the api.
-	Clock clock.Configuration `yaml:"clock"`
 }
 
-// R2StoreConfiguration has all the fields necessary for an R2Store
-type R2StoreConfiguration struct {
+// r2StoreConfiguration has all the fields necessary for an R2 store.
+type r2StoreConfiguration struct {
```
Comment (dgromov) [inline services/r2ctl/config/r2ctl.go:64]: so if anyone wants to run a different main, they have to completely redefine all of these configs?

File: services/r2ctl/main/main.go [hunk 2]
```diff
@@ -90,12 +89,10 @@ func main() {
 		SetMetricsSamplingRate(cfg.Metrics.SampleRate()).
 		SetReportInterval(cfg.Metrics.ReportInterval())
 
-	clockOpts := clock.NewOptions()
-
 	listenAddr := fmt.Sprintf("%s:%d", cfg.HTTP.Host, cfg.HTTP.Port)
 	serverOpts := cfg.HTTP.NewHTTPServerOptions(instrumentOpts)
 
-	store, err := cfg.Store.NewR2Store(instrumentOpts, clockOpts)
```
Comment (dgromov) [inline services/r2ctl/main/main.go:98]: what happened to the clockOpts?

File: services/r2ctl/config/r2ctl.go [hunk 3]
```diff
@@ -52,31 +54,41 @@ type Configuration struct {
...
-	// KV namespace.
-	RulesKVConfig clusterKV.Configuration `yaml:"rulesKVConfig"`
+// kvStoreConfig is the configuration for the kv backed implementation of the R2 store.
+type kvStoreConfig struct {
+	// KVClient configures the client for key value store.
+	KVClient *etcd.Configuration `yaml:"kvClient" validate:"nonzero"`
 
-	// KVClient configuration for key value store.
-	KVClient *etcd.Configuration `yaml:"etcd" validate:"nonzero"`
+	// KV configures the namespace and the environment.
```
Comment (dgromov) [inline services/r2ctl/config/r2ctl.go]: s/KV/KVConfig/

File: services/r2ctl/config/r2ctl.go (hunk 1 above)
Comment (xichen2020) [reply services/r2ctl/config/r2ctl.go:64]: The idea is that the users wouldn't need to run their own main file if the config here is flexible enough, and I'd rather keep it that way.

File: services/r2ctl/config/r2ctl.go (hunk 3 above)
Comment (xichen2020) [reply services/r2ctl/config/r2ctl.go]: Sure sounds good.

File: services/r2ctl/main/main.go (hunk 2 above)
Comment (xichen2020) [reply services/r2ctl/main/main.go:98]: It's initialized to `clock.NewOptions()` by default and it's primarily useful to testing. I don't imagine we'll need to configure it in the config, and if we do, we can always add them back.

**Review by dgromov — APPROVED** [review]

**Pull request merged**